src/qibolab/_core/instruments/emulator/emulator.py

```diff
@@ -208,7 +231,7 @@ def hamiltonian(
     n = hamiltonian.transmon_levels
     op = expand(config.operator(n), hamiltonian.dims, qubit)
     waveforms = (
-        waveform(pulse, config)
+        waveform(pulse, config, hamiltonian.single_qubit[qubit].detuned_frequency)
```

Comment:
With the solution proposed above, you may want to pass the whole `hamiltonian` (or `hamiltonian.single_qubit[qubit]`), instead of the individual attribute.

Reply:
If you prefer not pass the attribute then I will need to pass `hamiltonian.single_qubit[qubit]`. Otherwise I need to pass both `hamiltonian` and `qubit`.